environment/src/main/java/com/sequenceiq/environment/network/dao/domain/GcpNetwork.java

```diff
@@ -48,13 +67,41 @@ public String getNetworkId() {
         return networkId;
     }
 
+    public Json getZoneMetas() {
+        return zoneMetas;
+    }
+
+    public void setZoneMetas(Json zoneMetas) {
+        this.zoneMetas = zoneMetas;
+    }
+
+    public Set<String> getAvailabilityZones() {
+        Set<String> zoneList = new HashSet<>();
+        Json zoneMetasJson = getZoneMetas();
+        if (zoneMetasJson != null) {
+            zoneList.addAll((List<String>) zoneMetasJson
+                    .getMap()
+                    .getOrDefault(NetworkConstants.AVAILABILITY_ZONES, new ArrayList<>()));
+        }
+        return zoneList;
+    }
+
+    public void setAvailabilityZones(Set<String> zones) {
+        if (CollectionUtils.isNotEmpty(zones)) {
+            Map<String, Object> existingAttributes = (zoneMetas != null) ? zoneMetas.getMap() : new HashMap<>();
+            existingAttributes.put(NetworkConstants.AVAILABILITY_ZONES, zones);
+            setZoneMetas(new Json(existingAttributes));
+        }
+    }
```

Comment:
We can move logic of converting json to Set and vice versa to BaseNetwork so that it can be used by GCP as well as Azure.